plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -82,21 +82,18 @@ func getZoneKey(node *api.Node) string {
 // pods which match the same service selectors or RC selectors as the pod being scheduled.
 // Where zone information is included on the nodes, it favors nodes in zones with fewer existing matching pods.
 func (s *SelectorSpread) CalculateSpreadPriority(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
-	selectors := make([]labels.Selector, 0)
-	services, err := s.serviceLister.GetPodServices(pod)
-	if err == nil {
+	selectors := make([]labels.Selector, 0, 3)
```

Comment:
Generally we expect no more than 1 service, no more than 1 RC and no more than 1 RS. (Probably even 2 in heapthy case is enough (either RC or RS)). But this doesn't really change anything...